src/viewer.js

```diff
@@ -373,6 +376,19 @@ $.Viewer = function( options ) {
         if (!_this._updateRequestId) {
             _this._updateRequestId = scheduleUpdate( _this, updateMulti );
         }
+
+        // Attaches a handler to monitor individual TiledImage loading states
+        // propagate aggregate viewer-level fully-loaded status changes.
+        var tiledImage = event.item;
+        tiledImage.addHandler('fully-loaded-change', function() {
```

Comment:
We should remove this event handler when the `remove-item` handler below gets called. It may not be entirely necessary, but it's good hygiene.